packages/vite/src/node/plugins/worker.ts

```diff
@@ -55,17 +62,25 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
       if (isBuild) {
         // bundle the file as entry to support imports
         const rollup = require('rollup') as typeof Rollup
+        const { plugins, rollupOptions, format } = workerBundleOption
+        const [prePlugins, normalPlugins, postPlugins] = plugins
         const bundle = await rollup.rollup({
           input: cleanUrl(id),
-          plugins: await resolvePlugins({ ...config }, [], [], []),
+          plugins: await resolvePlugins(
+            { ...config },
+            prePlugins,
+            normalPlugins,
+            postPlugins
+          ),
           onwarn(warning, warn) {
             onRollupWarning(warning, warn, config)
-          }
+          },
+          ...rollupOptions
```

Comment:
rollupOptions plugins should have less precedence than Vite options, check other places where this is used here https://github.com/vitejs/vite/blob/1f945f62bf4a722c95a7b8f9c14c32a6f2be5c3f/packages/vite/src/node/build.ts#L461 

In `resolveBuildPlugins`, the plugins from the general `rollupOptions` are inserted at the end. We should add an option to resolve with the same rule from `worker.rollupOptions`: https://github.com/vitejs/vite/blob/1f945f62bf4a722c95a7b8f9c14c32a6f2be5c3f/packages/vite/src/node/build.ts#L361
This is a breaking change. I don't expect it would affect anyone, but let's do this in the 2.8 beta

Reply:
I am omit it from dts😅